### ISSUE TITLE
[$1000] Workspace - No correct error message when entering a workspace link without access

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23283]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to this link https://staging.new.expensify.com/workspace/77336CD9870F8DB1
(does not have any users)
2. Log in

## Expected Result:
"You do not have access to this page. Are you trying to join the workspace? Please reach out to the owner of this workspace so they can add you as a member! Something else? Reach out to Concierge" error message should be shown.

## Actual Result:
"Hmm... it's not here. Oops, this page cannot be found" error message is shown.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.43-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/a9d44bcf-c319-4405-b7c2-58e37125c335

**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010da14427ae7a9920</li>
        <li>Upwork Job ID: 1682126495724613632</li>
        <li>Last Price Increase: 2023-07-20</li>
    </ul>
</details>

**Comment by Nikhil-Vats** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace - No correct error message when entering a workspace link without access

### What is the root cause of that problem?
Here, the subtitle passed is wrong -
https://github.com/Expensify/App/blob/7fea760c2bb7556815aaca2d61861bd1cf39d6b1/src/pages/workspace/WorkspaceInitialPage.js#L188

According to the condition, `notAuthorized` message will be shown when policy is not empty. This is wrong since if user doesn't have access then policy will be empty and `notAuthorized` message will not be shown.
### What changes do you think we should make in order to solve the problem?
Right now, there is no way to know if policy/workspace actually exists or if the user doesn't have access. Unless there is some data returned by backend which tells if policy even exists or not we can't properly decide whether to show the unauthorized message or unexisting policy message.

So, what we can do in the meantime is to always show `notAuthorized` message by changing line 188 to -
```javascript
subtitleKey='workspace.common.notAuthorized'
```

We have to do the same change on following pages since this problem exists on all workspace pages -
1. WorkspaceInitialPage
2. WorkspaceSettingsPage
3. WorkspaceCardPage
4. WorkspaceReimbursePage
5. WorkspaceBillsPage
6. WorkspaceInvoicesPage
7. WorkspaceTravelPage
8. WorkspaceMembersPage
9. ReimbursementAccountPage

### What alternative solutions did you explore? (Optional)
NA

**Comment by mehvash-Khan** [comment]
I don't think this is a bug and this behaviour is expected. `Hmm... it's not here. Oops, this page cannot be found` error message is being displayed for the user who is trying to access workspace link he/she is not a part of. Since the user is not a part of that workspace at all the backend does not return any data for this policy and hence `props.policy` is empty giving this error. Error `You do not have access to this page. Are you trying to join the workspace?.......` error should be displayed in the case where user is trying to access workspace link he is a part of.

**Comment by dhanashree-sawant** [comment]
In case if it is taken up as bug moving forward, #23188 is similar and was raised before, can we change or add my name in reporting?

**Comment by yh-0218** [comment]
## Proposal

The issue #23188 is the same but it's closed. So bring the proposal here

### Please re-state the problem that we are trying to solve in this issue.
Wrong error message when members try to open workspace link
### What is the root cause of that problem?
We set wrong `subtitleKey` prop here. If `props.policy` is empty that means the user isn't a member of the workspace, we should show `workspace.common.notAuthorized` error. 


https://github.com/Expensify/App/blob/e19a620d50b34029ff9c0733560f43e391c21432/src/pages/workspace/WorkspaceInitialPage.js#L188

And if not and the user isn't admin of the workspace we should display another error like `You don't have permission to access this workspace because you are just a member of the workspace, not the admin of the workspace`

This issue also occurs in other workspace pages: `WorkspaceInviteMessagePage`, `WorkspaceInvitePage`, `WorkspaceMembersPage`, `ReimbursementAccountPage`, 
### What changes do you think we should make in order to solve the problem?
We should create another translation key for the case user is a member of the workspace. And we should change the condition here like `!_.isEmpty(props.policy) ? undefined : 'workspace.common.notAuthorized'`

https://github.com/Expensify/App/blob/e19a620d50b34029ff9c0733560f43e391c21432/src/pages/workspace/WorkspaceInitialPage.js#L188
### What alternative solutions did you explore? (Optional)

**Comment by lschurr** [comment]
@thesahindia could you review the proposals here?

**Comment by lschurr** [comment]
>we already have an old issue https://github.com/Expensify/App/issues/12616.

So should this one be closed @thesahindia?

**Comment by kbecciv** [comment]
@lschurr We have TR steps https://expensify.testrail.io/index.php?/cases/view/2638169 for the above issue, can you please send the TR update issue if it's expected now.

**Comment by lschurr** [comment]
This is a very old issue, so I'm not sure about the expected behavior. Please start a conversation in Slack so the team can weigh in @kbecciv
